From 2efb785efb96b01de392ba41e5a3d1568c9c44c3 Mon Sep 17 00:00:00 2001
From: Kenny Gong <kenny.gong@intel.com>
Date: Tue, 24 Jul 2018 09:05:24 +0800
Subject: [PATCH] sleep 5s to wait for unlock user operation

add title for CarBleTrustAgent
correct settingsActivity for CarBleTrustAgent

Change-Id: I000f8a48b3c6bb98433ff1a7af82d6654685208d
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-66998
Signed-off-by: Kenny Gong <kenny.gong@intel.com>
---
 TrustAgent/res/xml/car_sample_trust_agent.xml              | 3 ++-
 TrustAgent/src/com/android/car/trust/CarBleTrustAgent.java | 6 ++++++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/TrustAgent/res/xml/car_sample_trust_agent.xml b/TrustAgent/res/xml/car_sample_trust_agent.xml
index 0c7d91d..0851850 100644
--- a/TrustAgent/res/xml/car_sample_trust_agent.xml
+++ b/TrustAgent/res/xml/car_sample_trust_agent.xml
@@ -16,5 +16,6 @@
   -->
 <trust-agent xmlns:android="http://schemas.android.com/apk/res/android"
              xmlns:priv-android="http://schemas.android.com/apk/prv/res/android"
-             android:settingsActivity=".MainActivity"
+             android:title="CarBleTrustAgent"
+             android:settingsActivity=".CarEnrolmentActivity"
              priv-android:unlockProfile="true" />
diff --git a/TrustAgent/src/com/android/car/trust/CarBleTrustAgent.java b/TrustAgent/src/com/android/car/trust/CarBleTrustAgent.java
index 227ea89..7165038 100644
--- a/TrustAgent/src/com/android/car/trust/CarBleTrustAgent.java
+++ b/TrustAgent/src/com/android/car/trust/CarBleTrustAgent.java
@@ -69,6 +69,12 @@ public class CarBleTrustAgent extends TrustAgentService {
                     + " Time: " + System.currentTimeMillis());
             unlockUserWithToken(handle, token, userHandle);
 
+            // sleep 5s to wait for user unlock operation
+            try {
+                Thread.sleep(5000);
+            } catch (InterruptedException e) {
+            }
+
             Log.d(TAG, "Attempted to unlock user, is user unlocked: "
                     + um.isUserUnlocked(userHandle)
                     + " Time: " + System.currentTimeMillis());
-- 
2.7.4

